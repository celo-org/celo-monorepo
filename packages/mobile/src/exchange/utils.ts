import BigNumber from 'bignumber.js'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { convertToContractDecimals } from 'src/tokens/saga'
import { getContractKitAsync } from 'src/web3/contracts'

export async function celoTocUsd(celoAmount: BigNumber) {
  const celoAmountInWei = (
    await convertToContractDecimals(celoAmount, CURRENCY_ENUM.GOLD)
  ).integerValue()

  const contractKit = await getContractKitAsync()
  const exchange = await contractKit.contracts.getExchange()
  const exchangeRate = await exchange.getGoldExchangeRate(celoAmountInWei)
  const cUsdAmount = exchangeRate.multipliedBy(celoAmount)

  return cUsdAmount
}
