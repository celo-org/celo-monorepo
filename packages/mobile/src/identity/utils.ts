import BigNumber from 'bignumber.js'

export function isUserBalanceSufficient(
  userBalance: string | number | null,
  estimatedTxAmount: number | string
) {
  if (!userBalance || new BigNumber(userBalance).isLessThan(new BigNumber(estimatedTxAmount))) {
    return false
  }

  return true
}
