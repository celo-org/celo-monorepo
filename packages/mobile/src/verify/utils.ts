import { parsePhoneNumber } from '@celo/utils/lib/phoneNumbers'

export function getPhoneNumberState(
  phoneNumber: string,
  countryCallingCode: string,
  countryCodeAlpha2: string
) {
  const phoneDetails = parsePhoneNumber(phoneNumber, countryCallingCode)

  if (phoneDetails) {
    return {
      // Show international display number to avoid confusion
      internationalPhoneNumber: phoneDetails.displayNumberInternational,
      e164Number: phoneDetails.e164Number,
      isValidNumber: true,
      countryCodeAlpha2: phoneDetails.regionCode!,
    }
  } else {
    return {
      internationalPhoneNumber: phoneNumber,
      e164Number: '',
      isValidNumber: false,
      countryCodeAlpha2,
    }
  }
}
