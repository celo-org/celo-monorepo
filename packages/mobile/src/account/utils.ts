import { ADDRESS_LENGTH } from 'src/exchange/reducer'

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}
