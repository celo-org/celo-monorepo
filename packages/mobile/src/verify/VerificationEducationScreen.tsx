import Button, { BtnTypes } from '@celo/react-components/components/Button'
import Link from '@celo/react-components/components/Link'
import TextButton from '@celo/react-components/components/TextButton'
import VerifyPhone from '@celo/react-components/icons/VerifyPhone'
import colors from '@celo/react-components/styles/colors'
import fontStyles from '@celo/react-components/styles/fonts'
import { componentStyles } from '@celo/react-components/styles/styles'
import * as React from 'react'
import { withNamespaces, WithNamespaces } from 'react-i18next'
import { ScrollView, StyleSheet, Text, View } from 'react-native'
import Modal from 'react-native-modal'
import SafeAreaView from 'react-native-safe-area-view'
import { connect } from 'react-redux'
import componentWithAnalytics from 'src/analytics/wrapper'
import { Namespaces } from 'src/i18n'
import { setHasSeenVerificationNux } from 'src/identity/actions'
import { nuxNavigationOptions } from 'src/navigator/Headers'
import { navigate, navigateHome } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'

interface DispatchProps {
  setHasSeenVerificationNux: typeof setHasSeenVerificationNux
}

type Props = WithNamespaces & DispatchProps

interface State {
  isModalVisible: boolean
}

const mapDispatchToProps = {
  setHasSeenVerificationNux,
}

class VerificationEducationScreen extends React.Component<Props, State> {
  static navigationOptions = nuxNavigationOptions

  state: State = {
    isModalVisible: false,
  }

  onPressLearnMore = () => {
    navigate(Screens.VerificationLearnMoreScreen)
  }

  onPressStart = () => {
    this.props.setHasSeenVerificationNux(true)
    navigate(Screens.VerificationLoadingScreen)
  }

  onPressSkip = () => {
    this.setState({ isModalVisible: true })
  }

  onPressSkipCancel = () => {
    this.setState({ isModalVisible: false })
  }

  onPressSkipConfirm = () => {
    this.props.setHasSeenVerificationNux(true)
    navigateHome()
  }

  render() {
    const { t } = this.props
    return (
      <SafeAreaView style={styles.container}>
        <ScrollView contentContainerStyle={styles.scrollContainer}>
          <VerifyPhone />
          <Text style={styles.h1} testID="VerificationEducationHeader">
            {t('education.header')}
          </Text>
          <Text style={styles.body}>{t('education.body1')}</Text>
          <Text style={styles.body}>{t('education.body2')}</Text>
          <Link onPress={this.onPressLearnMore}>{t('education.learnMore')}</Link>
        </ScrollView>
        <>
          <Button
            text={t('education.start')}
            onPress={this.onPressStart}
            standard={false}
            type={BtnTypes.PRIMARY}
            testID="VerificationEducationContinue"
          />
          <Button
            text={t('education.skip')}
            onPress={this.onPressSkip}
            standard={false}
            type={BtnTypes.SECONDARY}
            testID="VerificationEducationSkip"
          />
        </>
        <Modal isVisible={this.state.isModalVisible}>
          <View style={styles.modalContainer}>
            <Text style={styles.modalHeader}>{t('skipModal.header')}</Text>
            <Text style={fontStyles.body}>{t('skipModal.body1')}</Text>
            <Text style={[fontStyles.body, componentStyles.marginTop10]}>
              {t('skipModal.body2')}
            </Text>
            <View style={styles.modalButtonsContainer}>
              <TextButton onPress={this.onPressSkipCancel} style={styles.modalCancelText}>
                {t('global:cancel')}
              </TextButton>
              <TextButton onPress={this.onPressSkipConfirm} style={styles.modalSkipText}>
                {t('global:skip')}
              </TextButton>
            </View>
          </View>
        </Modal>
      </SafeAreaView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: colors.background,
  },
  scrollContainer: {
    flex: 1,
    padding: 30,
    paddingTop: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  h1: {
    ...fontStyles.h1,
    marginTop: 20,
  },
  body: {
    ...fontStyles.bodyLarge,
    textAlign: 'center',
    marginBottom: 20,
  },
  modalContainer: {
    backgroundColor: colors.background,
    padding: 20,
    marginHorizontal: 10,
    borderRadius: 4,
  },
  modalHeader: {
    ...fontStyles.h2,
    ...fontStyles.bold,
    marginVertical: 15,
  },
  modalButtonsContainer: {
    marginTop: 25,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
  modalCancelText: {
    ...fontStyles.body,
    ...fontStyles.semiBold,
    paddingRight: 20,
  },
  modalSkipText: {
    ...fontStyles.body,
    ...fontStyles.semiBold,
    color: colors.celoGreen,
    paddingLeft: 20,
  },
})

export default componentWithAnalytics(
  connect<{}, DispatchProps>(
    null,
    mapDispatchToProps
  )(withNamespaces(Namespaces.nuxVerification2)(VerificationEducationScreen))
)
