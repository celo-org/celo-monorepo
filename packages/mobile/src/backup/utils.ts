import { sampleSize } from 'lodash'
import { AsyncStorage } from 'react-native'
import { generateMnemonic, wordlists } from 'react-native-bip39'
import { getKey, setKey } from 'src/utils/keyStore'
import Logger from 'src/utils/Logger'

const TAG = 'Backup/utils'

export async function createQuizWordList(mnemonic: string, language: string | null) {
  const disallowedWordSet = new Set(mnemonic.split(' '))
  const languageWordList = getWordlist(language)
  const wordOptions: string = await generateMnemonic(1000, null, languageWordList)
  const quizWordList = new Set(
    [...wordOptions.split(' ')].filter((word: string) => !disallowedWordSet.has(word))
  )
  return [...quizWordList]
}

export function selectQuizWordOptions(correctWord: string, allWords: string[], numOptions: number) {
  const wordOptions = []
  const correctWordPosition = Math.floor(Math.random() * numOptions)
  const randomWordIndexList = sampleSize([...Array(allWords.length).keys()], numOptions - 1)
  let randomWordIndex: number = 0

  for (let i = 0; i < numOptions; i++) {
    if (i === correctWordPosition) {
      wordOptions.push(correctWord)
      continue
    }

    wordOptions.push(allWords[randomWordIndexList[randomWordIndex]])
    randomWordIndex += 1
  }
  return wordOptions
}

export function getWordlist(language: string | null) {
  if (!language) {
    return wordlists.EN
  }

  switch (language.slice(0, 2)) {
    case 'es': {
      return wordlists.ES
    }
    default: {
      return wordlists.EN
    }
  }
}

// TODO(Rossy) Remove after the next alfa testnet reset
export async function getStoredMnemonic(): Promise<string | null> {
  try {
    Logger.debug(TAG, 'Checking keystore for mnemonic')
    let mnemonic = await getKey('mnemonic')
    if (mnemonic) {
      return mnemonic
    }

    Logger.debug(TAG, 'Mnemonic not found in keystore, checking async storage')
    mnemonic = await AsyncStorage.getItem('mnemonic')
    if (mnemonic) {
      await setKey('mnemonic', mnemonic)
      await AsyncStorage.removeItem('mnemonic')
      return mnemonic
    }

    Logger.error(TAG, 'No mnemonic found')
    return null
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}
