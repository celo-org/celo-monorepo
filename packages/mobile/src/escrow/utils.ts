import { ensureLeading0x } from '@celo/base'
import { generateDeterministicInviteCode } from '@celo/utils/lib/account'
import { publicKeyToAddress } from '@celo/utils/lib/address'

export const generateEscrowPaymentIdAndPk = (
  recipientPhoneHash: string,
  recipientPepper: string,
  addressIndex: number = 0
) => {
  const { publicKey, privateKey } = generateDeterministicInviteCode(
    recipientPhoneHash,
    recipientPepper,
    addressIndex
  )
  return { paymentId: publicKeyToAddress(publicKey), privateKey: ensureLeading0x(privateKey) }
}

export const generateUniquePaymentId = (
  existingPaymentIds: string[],
  phoneHash: string,
  pepper: string
) => {
  const paymentIdSet: Set<string> = new Set(existingPaymentIds)

  // Using an upper bound of 100 to be sure this doesn't run forever given
  // the realistic amount of pending escrow txs is far less than this
  for (let i = 0; i < 100; i += 1) {
    const { paymentId } = generateEscrowPaymentIdAndPk(phoneHash, pepper, i)
    if (!paymentIdSet.has(paymentId)) {
      return paymentId
    }
  }
}
