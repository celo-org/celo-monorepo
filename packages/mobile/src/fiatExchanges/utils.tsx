import { SIMPLEX_URI } from 'src/config'
import { CURRENCY_ENUM } from 'src/geth/consts'
import { LocalCurrencyCode } from 'src/localCurrency/consts'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { navigateToURI } from 'src/utils/linking'

export const openMoonpay = (currencyCode: LocalCurrencyCode, currencyToBuy: CURRENCY_ENUM) => {
  navigate(Screens.MoonPay, {
    localAmount: 0,
    currencyCode,
    currencyToBuy,
  })
}

export const openSimplex = (account: string | null) => {
  navigateToURI(`${SIMPLEX_URI}?address=${account}`)
}
