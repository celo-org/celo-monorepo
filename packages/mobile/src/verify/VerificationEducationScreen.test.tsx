import * as React from 'react'
import 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import { Provider } from 'react-redux'
import VerificationEducationScreen from 'src/verify/VerificationEducationScreen'
import { createMockStore } from 'test/utils'

describe('VerificationEducationScreen', () => {
  const store = createMockStore({})

  it('renders correctly', () => {
    const { getByTestId, toJSON } = render(
      <Provider store={store}>
        <VerificationEducationScreen />
      </Provider>
    )
    expect(toJSON()).toMatchSnapshot()

    // And snapshot again after showing the modal
    const skipButton = getByTestId('VerificationEducationSkip')
    fireEvent.press(skipButton)
    expect(toJSON()).toMatchSnapshot()
  })
})
