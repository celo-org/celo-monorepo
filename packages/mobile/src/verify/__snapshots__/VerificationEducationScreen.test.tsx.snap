// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerificationEducationScreen renders correctly 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 30,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 20,
            },
            Object {},
          ]
        }
        testID="VerificationEducationIcon"
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../packages/mobile/src/images/coins-logo.png",
            }
          }
          style={
            Object {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Light",
            "fontSize": 22,
            "marginTop": 20,
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
        testID="VerificationEducationHeader"
      >
        education.header
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 17,
            "lineHeight": 26,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        education.body1
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 17,
            "lineHeight": 26,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        education.body2
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackgroundBorderless",
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 13,
                "lineHeight": 18,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          education.learnMore
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 3,
          "flexDirection": "row",
        },
        null,
        null,
        undefined,
        Object {
          "backgroundColor": "#42D689",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      nativeBackgroundAndroid={
        Object {
          "attribute": "selectableItemBackground",
          "type": "ThemeAttrAndroid",
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#42D689",
          },
          Object {
            "height": 50,
          },
          Object {
            "borderWidth": 0,
          },
        ]
      }
      testID="VerificationEducationContinue"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Hind-SemiBold",
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "paddingLeft": 5,
                "paddingRight": 5,
              },
            ]
          }
        >
          education.start
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 3,
          "flexDirection": "row",
        },
        null,
        null,
        undefined,
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      nativeBackgroundAndroid={
        Object {
          "attribute": "selectableItemBackground",
          "type": "ThemeAttrAndroid",
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 50,
          },
          Object {
            "borderWidth": 0,
          },
        ]
      }
      testID="VerificationEducationSkip"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Hind-SemiBold",
                "fontSize": 16,
              },
              Object {
                "color": "#42D689",
              },
              Object {
                "paddingLeft": 5,
                "paddingRight": 5,
              },
            ]
          }
        >
          education.skip
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Light",
              "fontSize": 22,
              "paddingBottom": 20,
              "textAlign": "center",
            }
          }
        >
          skipModal.header
        </Text>
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
        >
          skipModal.body1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          skipModal.body2
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginTop": 20,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackgroundBorderless",
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "color": "#2E3338",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingRight": 20,
                  },
                ]
              }
            >
              global:cancel
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackgroundBorderless",
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingLeft": 20,
                  },
                ]
              }
            >
              global:skip
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</SafeAreaView>
`;

exports[`VerificationEducationScreen renders correctly 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 30,
        "paddingTop": 0,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginBottom": 20,
            },
            Object {},
          ]
        }
        testID="VerificationEducationIcon"
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../packages/mobile/src/images/coins-logo.png",
            }
          }
          style={
            Object {
              "height": 35,
              "width": 35,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Light",
            "fontSize": 22,
            "marginTop": 20,
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
        testID="VerificationEducationHeader"
      >
        education.header
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 17,
            "lineHeight": 26,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        education.body1
      </Text>
      <Text
        style={
          Object {
            "color": "#2E3338",
            "fontFamily": "Hind-Regular",
            "fontSize": 17,
            "lineHeight": 26,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        education.body2
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackgroundBorderless",
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 13,
                "lineHeight": 18,
                "textDecorationLine": "underline",
              },
              undefined,
            ]
          }
        >
          education.learnMore
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 3,
          "flexDirection": "row",
        },
        null,
        null,
        undefined,
        Object {
          "backgroundColor": "#42D689",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      nativeBackgroundAndroid={
        Object {
          "attribute": "selectableItemBackground",
          "type": "ThemeAttrAndroid",
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#42D689",
          },
          Object {
            "height": 50,
          },
          Object {
            "borderWidth": 0,
          },
        ]
      }
      testID="VerificationEducationContinue"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Hind-SemiBold",
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
              Object {
                "paddingLeft": 5,
                "paddingRight": 5,
              },
            ]
          }
        >
          education.start
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 3,
          "flexDirection": "row",
        },
        null,
        null,
        undefined,
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      nativeBackgroundAndroid={
        Object {
          "attribute": "selectableItemBackground",
          "type": "ThemeAttrAndroid",
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "height": 50,
          },
          Object {
            "borderWidth": 0,
          },
        ]
      }
      testID="VerificationEducationSkip"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Hind-SemiBold",
                "fontSize": 16,
              },
              Object {
                "color": "#42D689",
              },
              Object {
                "paddingLeft": 5,
                "paddingRight": 5,
              },
            ]
          }
        >
          education.skip
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Light",
              "fontSize": 22,
              "paddingBottom": 20,
              "textAlign": "center",
            }
          }
        >
          skipModal.header
        </Text>
        <Text
          style={
            Object {
              "color": "#2E3338",
              "fontFamily": "Hind-Regular",
              "fontSize": 16,
              "lineHeight": 24,
            }
          }
        >
          skipModal.body1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          skipModal.body2
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginTop": 20,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackgroundBorderless",
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "color": "#2E3338",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingRight": 20,
                  },
                ]
              }
            >
              global:cancel
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            nativeBackgroundAndroid={
              Object {
                "attribute": "selectableItemBackgroundBorderless",
                "type": "ThemeAttrAndroid",
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "color": "#42D689",
                    "fontFamily": "Hind-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingLeft": 20,
                  },
                ]
              }
            >
              global:skip
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</SafeAreaView>
`;
